fluent/fluentd
# in_http: Do not discard the 'time' field in incoming JSON records.

### What is the problem?

It was found that `parse_params_default()` unconditionally discards the `time`
value returned by the JSON parser. This trivially leads to silent data loss.

### How to reproduce the issue

We can confirm this by following these steps:

 1. Start a Fluentd server with settings like below:

    ```xml
    <source>
      @type http
    </source>
    <match debug.**>
      @type stdout
    </match>
    ```

 2. Send a HTTP request with the following command:

    ```bash
    $ curl -X POST -d 'json={"time":1}' http://localhost:8888/debug.log
    ```

 3. Now the original JSON data has been lost, leaving a empty hash with
    a timestamp of the current server time.

    ```
    2018-02-21 15:44:57.767980983 +0900 debug.log: {}
    ```

### Solution

This patch fixes the issue by passing the timestamp (extracted from
the incoming JSON data) to the caller.

With this patch applied, the `time` field in the JSON data will be properly 
reflected to the event data:

```
1970-01-01 09:00:01.000000000 +0900 debug.log: {}
```

### Note on the context of this patch

I've found this issue while updating the documentation of in_http
at https://github.com/fluent/fluentd-docs/pull/457.
